app/controllers/manageBusinesses/add/IncomeSourceCheckDetailsController.scala

```diff
@@ -169,8 +170,13 @@ class IncomeSourceCheckDetailsController @Inject()(val checkDetailsView: IncomeS
   private def handleSubmit(isAgent: Boolean, incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_]): Future[Result] = {
     withSessionData(IncomeSourceJourneyType(Add, incomeSourceType), BeforeSubmissionPage) { sessionData =>
 
-      val redirectUrl: (Boolean, IncomeSourceType) => String = (isAgent: Boolean, incomeSourceType: IncomeSourceType) =>
-        routes.IncomeSourceReportingMethodController.show(isAgent, incomeSourceType).url
+      val redirectUrl: (Boolean, IncomeSourceType) => String = (isAgent: Boolean, incomeSourceType: IncomeSourceType) =>{
+        if(isEnabled(IncomeSourcesNewJourney)){
+          controllers.manageBusinesses.add.routes.IncomeSourceReportingMethodController.show(isAgent, incomeSourceType).url
+        }else{
+          routes.IncomeSourceReportingMethodController.show(isAgent, incomeSourceType).url
```

Comment:
We should navigate them to home page in this case as he should never be using this url except in case of bookmarking it if the newJourney is disabled.